[jit][subgraph_matcher] Enable regex matching for string attributes of node

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39615 [quant][graphmode] Fix a corner case in handling `if` in insert_observers
* #39551 [quant][graphmode] Support propagate dequantize for nodes with multiple outputs
* #39550 [quant][graphmode][refactor] propagateQuantizationOps
* #39490 [quant][graphmode] Support conv*d_relu in traced models
* #39455 [quant][graphmode] Fix add_relu patterns for scripting and tracing
* **#39454 [jit][subgraph_matcher] Enable regex matching for string attributes of node**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21876224](https://our.internmc.facebook.com/intern/diff/D21876224)

jerryzh168 committed:
[jit][subgraph_matcher] Enable regex matching for string attributes of node

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit][subgraph_matcher] Enable regex matching for string attributes of node"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good, thanks for adding this!

jerryzh168 committed:
Update on "[jit][subgraph_matcher] Enable regex matching for string attributes of node"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit][subgraph_matcher] Enable regex matching for string attributes of node"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21876224](https://our.internmc.facebook.com/intern/diff/D21876224)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit][subgraph_matcher] Enable regex matching for string attributes of node"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21876224](https://our.internmc.facebook.com/intern/diff/D21876224)

[ghstack-poisoned]
